### ISSUE TITLE
package request: Matlab

### ISSUE THREAD

**Issue by tycho01** [NixOS/nixpkgs#56887]
## Issue description

https://www.mathworks.com/

**Comment by markuskowa** [comment]
I have tried to package it in the past but couldn't come up with a clean solution. Matlab is certainly a difficult program to package. I would be happy to review a PR if someone comes up with a workable solution.

**Comment by MetaDark** [comment]
I don't think this would ever be added to nixpkgs since there aren't any publicly available downloads and the trial version requires registering an account.

**Comment by Scriptkiddi** [comment]
I could be packaged with a requirement to prefetch the installation zip file, like its done for the displaylink package

**Comment by markuskowa** [comment]
There are several problems with Matlab:
* It needs to run an installer that pulls content online (won't work in a sandboxed derivation)
* An activation key or login credentials are required during the installation (that makes the derivation impure).
* Depending on which toolboxes are installed you may need different dependencies. Given the amount of toolboxes available makes it a very complex issue.

I am using an installed and activated version, that [lives outside the nix store](https://github.com/markuskowa/NixOS-QChem/blob/master/matlab/default.nix). This method works for me but is not a viable solution for inclusion in nixpkgs.

**Comment by Scriptkiddi** [comment]
still important to me

**Comment by Scriptkiddi** [comment]
still important to me

**Comment by Scriptkiddi** [comment]
@Mazurel you are completely right, that it would be difficult to package. But I would argue that matlab is widely used in the scientific community and as such it would be nice to allow those user to just add matlab to their system without dealing with flakes or overlays.

**Comment by arunoruto** [comment]
I am currently trying to run Maltab on NixOS and sure, having a package would be awesome, but is quite challenging. Maybe instead of using the installer, one could use [mpm](https://www.mathworks.com/help/install/ug/matlab-package-manager.html) or build a flake which works with mpm to make versioning easier.
Furthermore, even tho not elegant, one can run Matlab using a docker container on their system! Their ["guide"](https://hub.docker.com/r/mathworks/matlab) on the docker hub page is quite comprehensive, but look up the mathworks page and the GitHub page of the project. They support running is using X11 from the container, using your browser and VNC (browser and client).
Hope this helps someone while waiting for a better solution 🤙🏻

**Comment by SigmaSquadron** [comment]
Thanks for opening this package request. Unfortunately, in order to mitigate the unsustainable growth of unmaintained packages, Nixpkgs is no longer using package requests to coordinate new package additions.

That said, **we encourage you to [contribute](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#quick-start-to-adding-a-package) this package as a Pull Request yourself** - should you need any help during packaging, feel free to reach out on [Matrix](https://matrix.to/#/#dev:nixos.org) or [Discourse](https://discourse.nixos.org/c/dev/14). You are also welcome to open custom _Tracking_ issues if you would like the assistance of other volunteers in packaging a software ecosystem.

This change has the benefit of decreasing future maintainer burden on the long term, and was implemented after several contributors reached a consensus. You can read more about this on Pull Request [#391926](https://www.github.com/NixOS/nixpkgs/pull/391926) and Issue [#425040](https://www.github.com/NixOS/nixpkgs/issues/425040).
